### PR TITLE
Add testable prompt utilities with unit tests and updated documentation

### PR DESCRIPTION
**Pull request by DanielDTech**
## Summary

Refactors the prompt utility module (`confirm` and `prompt`) to accept injectable `input`/`output` stream options, making them fully testable. Adds comprehensive unit tests and updates the README with documentation for Prompt Utils and recently added `autoImport` options.

## Changes

- Refactor `confirm()` and `prompt()` to accept `{ input, output }` options instead of hardcoded stdin/stdout, enabling mock stream injection for testing
- Add 8 unit tests covering `confirm` (y/N behavior, case insensitivity, whitespace trimming, default rejection) and `prompt` (trimmed input, empty input)
- Document new Prompt Utils section in README with usage examples
- Document `recursive`, `recursiveNaming`, and `namePrefix` options for `autoImport` in README
- Add `camelCaseToKebabCase` to the README utility table
- Add `.claude/` project structure and improvements docs

### CODE REVIEW

File: .claude/improvements.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# Improvements
```
[the remark this answers is not in the document]
Comment (DanielDTech) [reply .claude/improvements.md]: Removed in 916a276.

File: .claude/project-structure.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
+# Project Structure
```
[the remark this answers is not in the document]
Comment (DanielDTech) [reply .claude/project-structure.md]: Added a table of contents index in 916a276.
